src/_pytest/compat.py

```diff
@@ -46,7 +45,7 @@
 # fmt: on
 
 
-def legacy_path(path: Union[str, "os.PathLike[str]"]) -> LEGACY_PATH:
+def legacy_path(path: str | os.PathLike[str]) -> LEGACY_PATH:
```

Comment:
Note: type annotations changed were introduced by `pyupgrade` because I added the `from __future__ import annotations` import.